recipes/boost/all/conanfile.py

```diff
@@ -1336,6 +1336,11 @@ def create_library_config(deps_name, name):
         ldflags = " ".join(self.conf.get("tools.build:sharedlinkflags", default=[], check_type=list)) + " "
         asflags = buildenv_vars.get("ASFLAGS", "") + " "
 
+        defines = " ".join(f"-D{d}" for d in self.conf.get("tools.build:defines", default=[], check_type=list)) + " "
+        if defines.strip():
+            cflags += defines
+            cxxflags += defines
```

Comment:
I'm not so sure, because b2 offers `define` as well: https://www.boost.org/doc/libs/1_84_0/tools/build/doc/html/index.html#bbv2.builtin.features.define

At same time, Conan config offers `tools.build:cxxflags` and `tools.build:cflags`

https://docs.conan.io/2/reference/config_files/global_conf.html

Reply:
I have tried adding the defines with:
```
        for d in self.conf.get("tools.build:defines", default=[], check_type=list):
            contents += f'<define>{d} '
```
But it does not work. I have no clue about b2 and its config files, so I have no chance of debugging it.

The user.config.jam produced with the above looks like this:
```
using zlib : 1.2.13 : <include>"[hidden]/zlib5c9e3f01b88a0/p/include" <search>"[hidden]/.conan2/p/b/zlib5c9e3f01b88a0/p/lib" <name>z ;
using "gcc" :  :  "[hidden]/.conan2/p/sdk-x4fa5888332018/p/gcc" :
<define>MYDEFINE1 <define>MYDEFINE2   ;
```
and the subsequent b2 call:
```
b2 -q architecture=x86 address-model=64 abi=sysv --layout=system --user-config=aaa/s/src/tools/build/user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=1 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=off --disable-iconv threading=multi visibility=hidden link=static variant=release --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-filesystem --with-graph --with-iostreams --with-math --with-program_options --with-random --with-regex --with-serialization --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=gcc cxxstd=14 define=_GLIBCXX_USE_CXX11_ABI=0 pch=on linkflags="" cxxflags="-fPIC" install --prefix=yyy -j16 --abbreviate-paths -d0 --debug-configuration --build-dir="xxx"
```

Reply:
And by the way, the `tools.build:cxxflags` and `tools.build:cflags` are already propagated to b2, I just added defines in my original PR.

Reply:
Are cflags and cxxflags not enough to your case? I'm missing what you are really trying to achieve with this change. 🤔

Reply:
@klausholstjacobsen Thank you for explaining! So only to be sure, you need to pass something like `gcc -DQNX` right? If correct, then `tools.build.defines` is the correct way in fact. I just need to check why it didn't work with b2 defines .

Reply:
That is absolutely correct!

Reply:
Why not reuse this?
https://github.com/conan-io/conan-center-index/blob/dec149b32f3a0670bf2395c0ccade27e7b760bca/recipes/boost/all/conanfile.py#L1074

Reply:
@stevenwdv Please, read the comment above: https://github.com/conan-io/conan-center-index/pull/22679#discussion_r1485671757

Reply:
@klausholstjacobsen Could please try the PR https://github.com/klausholstjacobsen/conan-center-index/pull/2 

I tested locally and worked for me. I tested in Mac and passed `-c "tools.build:defines=['_MSC_VER=191']"`, so it started activate all macros for Visual Studio and push new warning messages. I guess this is what you are looking for.

kudos to @stevenwdv

Reply:
Yeah I meant using `define=...` on the `b2` CLI instead of cflags/cxxflags or b2 config files

Reply:
@stevenwdv  Implemented your suggestion! Thanks!